<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>857</width>
    <height>436</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLineEdit" name="lineEdit1"/>
    </item>
    <item>
     <widget class="QLineEdit" name="lineEdit2"/>
    </item>
    <item>
     <widget class="QPushButton" name="concatenateButton">
      <property name="text">
       <string>拼接</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="reverseButton">
      <property name="text">
       <string>逆置</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="compareButton">
      <property name="text">
       <string>比较大小</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="substringButton">
      <property name="text">
       <string>求子串</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSpinBox" name="startSpinBox">
      <property name="maximum">
       <number>1000</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSpinBox" name="lengthSpinBox">
      <property name="maximum">
       <number>1000</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="resultLabel">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>concatenateButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_concatenateButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>43</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>42</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reverseButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_reverseButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>364</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>570</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compareButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_compareButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>476</x>
     <y>590</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>substringButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_substringButton_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>630</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_concatenateButton_clicked()</slot>
  <slot>on_reverseButton_clicked()</slot>
  <slot>on_compareButton_clicked()</slot>
  <slot>on_substringButton_clicked()</slot>
 </slots>
</ui>
